frontstage/views/secure_messaging/message_get.py

```diff
@@ -48,8 +49,13 @@ def view_conversation(session, thread_id):
             return redirect(url_for('secure_message_bp.view_conversation_list'))
 
     unread_message_count = {'unread_message_count': get_message_count_from_api(session)}
-    survey_name = get_survey(app.config['SURVEY_URL'], app.config['BASIC_AUTH'],
-                             refined_conversation[-1]['survey_id']).get('longName')
+    survey_name = None
+    try:
+        survey_name = get_survey(app.config['SURVEY_URL'], app.config['BASIC_AUTH'],
+                                 refined_conversation[-1]['survey_id']).get('longName')
+    except ApiError as exc:
+        logger.error('Failed to retrieve survey name', status_code=exc.status_code)
+
     business_name = conversation['messages'][-1]['@business_details']['name']
```

Comment:
I can't remember, but we're certain you'll always find a `@business_details -> name` in every message?  I think it throws a keyError if not which gives an ugly 500error screen so if there are any doubts then it needs to be handled as well

Reply:
Better to try-except then, especially if there's any chance of the error in future messages. Done, thanks.